parser/parser_test.go

```diff
@@ -634,3 +634,116 @@ states:
 		assert.Nil(t, workflow)
 	})
 }
+
+func TestUnmarshalWorkflowSwitchState(t *testing.T) {
+	t.Run("WorkflowSwitchStateEventConditions", func(t *testing.T) {
+		workflow, err := FromYAMLSource([]byte(`
+id: helloworld
+version: '1.0.0'
+specVersion: '0.8'
+name: Hello World Workflow
+description: Inject Hello World
+start: Hello State
+states:
+- name: Hello State
+  type: switch
+  eventConditions:
+  - eventRef: visaApprovedEvent
+    transition:
+      nextState: HandleApprovedVisa
+  - eventRef: visaRejectedEvent
+    transition:
+      nextState: HandleRejectedVisa
+  defaultCondition:
+    transition:
+      nextState: HandleNoVisaDecision
+- name: HandleApprovedVisa
+  type: operation
+  actions:
+  - subFlowRef:
+      workflowId: handleApprovedVisaWorkflowID
+  end:
+    terminate: true
+- name: HandleRejectedVisa
+  type: operation
+  actions:
+  - subFlowRef:
+      workflowId: handleRejectedVisaWorkflowID
+  end:
+    terminate: true
+- name: HandleNoVisaDecision
+  type: operation
+  actions:
+  - subFlowRef:
+      workflowId: handleNoVisaDecisionWorkfowId
+  end:
+    terminate: true
+
+`))
+		assert.Nil(t, err)
+		assert.NotNil(t, workflow)
+
+		b, err := json.Marshal(workflow)
+		assert.Nil(t, err)
+		assert.True(t, strings.Contains(string(b), "eventConditions"))
+
+		workflow = nil
+		err = json.Unmarshal(b, &workflow)
+		assert.Nil(t, err)
+	})
+
+	t.Run("WorkflowSwitchStateDataConditions", func(t *testing.T) {
+		workflow, err := FromYAMLSource([]byte(`
+id: helloworld
+version: '1.0.0'
+specVersion: '0.8'
+name: Hello World Workflow
+description: Inject Hello World
+start: Hello State
+states:
+- name: Hello State
+  type: switch
+  dataConditions:
```

Comment:
What happens if the dataConditions is spelled wrongly?

Reply:
It will report an error with `must have one of dataCnoditions, eventConditions`

Reply:
Cool, do you mind adding a step for this check?C

Reply:
sure, PTAL